src/pthread_httpGetRequest.c

```diff
@@ -0,0 +1,127 @@
+#include <stdio.h>  /* printf, sprintf */
+#include <stdlib.h> /* exit */
+#include <string.h> /* memcpy, memset */
+#include <unistd.h> /* read, write, close */
+
+#include <sys/types.h>
+#include <sys/socket.h> /* socket, connect */
+#include <netinet/in.h> /* struct sockaddr_in, struct sockaddr */
+#include <netdb.h>      /* struct hostent, gethostbyname */
+
+#include <pthread.h>
+
+// Method that calls perror method and print out the error
+void error(const char *msg)
+{
+  perror(msg);
+  exit(0);
+}
+
+// Buffer to store the respons message
+#define BUFFER_SIZE 6387162
+
+//Number of threads to be created
+#define NUM_THREADS     5
+
+//struct to pass args to mainOriginal from pthread_create()
+struct arg_struct {
+  int argc;
+  char **argv;
+};
+
+int mainOriginal(void *arguments)
+{
+  struct arg_struct *args = arguments;
+  
+  int looperdooper = 0;
+  
+  while (looperdooper <= 3){
+      int sockfd, portno, n;
+      struct sockaddr_in serv_addr;
+      struct hostent *server;
+    
+      char buffer[BUFFER_SIZE];
+      if (args -> argc < 3)
+      {
+        fprintf(stderr, "usage %s hostname port\n", args -> argv[0]);
+        exit(0);
+      }
+      portno = atoi(args -> argv[2]);
+      sockfd = socket(AF_INET, SOCK_STREAM, 0);
+      if (sockfd < 0)
+        error("ERROR opening socket");
+    
+      server = gethostbyname(args -> argv[1]);
+      if (server == NULL)
+      {
+        fprintf(stderr, "ERROR, no such host\n");
+        exit(0);
+      }
+    
+      bzero((char *)&serv_addr, sizeof(serv_addr));
+      serv_addr.sin_family = AF_INET;
+      bcopy((char *)server->h_addr,
+            (char *)&serv_addr.sin_addr.s_addr,
+            server->h_length);
+      serv_addr.sin_port = htons(portno);
```

Comment:
we don't need to assign server address for each request. This should be done once.